### ISSUE TITLE
other.test_no_warnings flake

### ISSUE THREAD

**Issue by kripken** [emscripten-core/emscripten#19272]
It errorred today here:

https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8782276878557668369/+/u/Emscripten_testsuite__other_/stdout

```
======================================================================
FAIL: test_no_warnings (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\b\s\w\ir\cache\vpython\store\cpython-e6tenrlsesftvg6k08ajgkk7b4\contents\bin\Lib\unittest\case.py", line 60, in testPartExecutor
    yield
  File "C:\b\s\w\ir\cache\vpython\store\cpython-e6tenrlsesftvg6k08ajgkk7b4\contents\bin\Lib\unittest\case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "C:\b\s\w\ir\cache\vpython\store\cpython-e6tenrlsesftvg6k08ajgkk7b4\contents\bin\Lib\unittest\case.py", line 633, in _callTestMethod
    method()
  File "C:\b\s\w\ir\x\w\install\emscripten\test\test_other.py", line 7086, in test_no_warnings
    self.assertEqual(err, '')
  File "C:\b\s\w\ir\cache\vpython\store\cpython-e6tenrlsesftvg6k08ajgkk7b4\contents\bin\Lib\unittest\case.py", line 912, in assertEqual
    assertion_func(first, second, msg=msg)
  File "C:\b\s\w\ir\cache\vpython\store\cpython-e6tenrlsesftvg6k08ajgkk7b4\contents\bin\Lib\unittest\case.py", line 1292, in assertMultiLineEqual
    self.fail(self._formatMessage(msg, standardMsg))
  File "C:\b\s\w\ir\cache\vpython\store\cpython-e6tenrlsesftvg6k08ajgkk7b4\contents\bin\Lib\unittest\case.py", line 753, in fail
    raise self.failureException(msg)
AssertionError: 'cache:INFO: generating system asset: symb[234 chars]ok\n' != ''
- cache:INFO: generating system asset: symbol_lists/e84a5c208bc82e3609d5cf66c989c23716df145a.json... (this will be cached in "C:\b\s\w\ir\x\w\install\emscripten\cache\symbol_lists\e84a5c208bc82e3609d5cf66c989c23716df145a.json" for subsequent builds)
- cache:INFO:  - ok
```

That should be impossible as the test guards against it specifically:

https://github.com/emscripten-core/emscripten/blob/dc4d654d4021d32cd85cc15d4ab7a01a92f6fac2/test/test_other.py#L7081-L7086

I can't reproduce a problem locally even if I clear the cache right before, which shows the guards to do work properly.

The only other idea I can think of is a race condition, and indeed it looks like the last line here will delete cache entries in one process, perhaps leading another to need to recreate them:

https://github.com/emscripten-core/emscripten/blob/dc4d654d4021d32cd85cc15d4ab7a01a92f6fac2/emcc.py#L545-L565

Not deleting there would be wasteful, so I'm not sure what we can do here. @sbc100 - thoughts?

**Comment by dschuff** [comment]
I just noticed this again. If you had a change like #19085 (even if it didn't change the default log level) you could use it to suppress the info notices per-test for tests that care about stderr (hopefully there aren't many of those).

**Comment by sbc100** [comment]
I believe this was fixed in #19531
